bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -707,8 +707,7 @@ public void processResult(int rc, String s, Object obj) {
      */
     public void shutdown() {
         LOG.info("Shutting down auditor");
-        submitShutdownTask();
-
+        executor.shutdown();
```

Comment:
wondering why did you make this change? why are you replacing submitShutdownTask call?

Reply:
I explained this in the description:

"the auditor shutdown logic is problematic. most of the tests can finish quickly however it spend more 30 seconds on shutting down.
because the shutdown logic will be blocked until awaitTermination timed out." 

if Auditor is running, it might be blocking in the thread, submit shutdown task will never be executed, then it will have to wait for 30 seconds timeout (`awaitTermination`) and issue `shutdownNow`. this change issues shutdown, which will interrupt the auditor task and shutdown the executor thread. 

Hope this make things clear

Reply:
yeah I get that, but I'm wondering why it wasn't like this before? why did the original author - https://github.com/apache/bookkeeper/commit/e5b0dd0d9e47394202f5a20cedf409f487d90beb went through this extra round of hassle for shutdown deliberately. Is it ok now to not to have this extra scrutiny with the shutdown process?

Reply:
No idea why it was done in that way. @ivankelly can chime in since he made that change.

Reply:
Long time ago, but I think it was to avoid killing a running audits before it was done, since it's a long running task.

Reply:
Okay so the question is can we change it to shutdown. Because even we don’t shutdown, it will be shutdown in 30 seconds.

Reply:
@ivankelly can you check if it is okay for us to change it to shutdown?